packages/waku/src/lib/handlers/dev-worker-api.ts

```diff
@@ -126,67 +125,30 @@ export async function renderRscWithWorker<Context>(
   const id = nextId++;
   const pipe = async () => {
     if (rr.stream) {
-      const reader = rr.stream.getReader();
-      try {
-        let result: ReadableStreamReadResult<unknown>;
-        do {
-          result = await reader.read();
-          if (result.value) {
-            const buf = result.value;
-            let mesg: MessageReq;
-            if (buf instanceof ArrayBuffer) {
-              mesg = { id, type: 'buf', buf, offset: 0, len: buf.byteLength };
-            } else if (buf instanceof Uint8Array) {
-              mesg = {
-                id,
-                type: 'buf',
-                buf: buf.buffer,
-                offset: buf.byteOffset,
-                len: buf.byteLength,
-              };
-            } else {
-              throw new Error('Unexepected buffer type');
-            }
-            worker.postMessage(mesg, [mesg.buf]);
-          }
-        } while (!result.done);
-      } catch (err) {
-        const mesg: MessageReq = { id, type: 'err', err };
-        worker.postMessage(mesg);
-      }
+      worker.postMessage(
+        { id, type: 'pipe', stream: rr.stream } as MessageReq,
+        [rr.stream as unknown as TransferListItem],
+      );
     }
-    const mesg: MessageReq = { id, type: 'end' };
-    worker.postMessage(mesg);
   };
   let started = false;
   return new Promise((resolve, reject) => {
-    let controller: ReadableStreamDefaultController<Uint8Array>;
-    const stream = new ReadableStream({
-      start(c) {
-        controller = c;
-      },
-    });
     messageCallbacks.set(id, (mesg) => {
       if (mesg.type === 'start') {
         if (!started) {
           started = true;
-          resolve([stream, mesg.context as Context]);
+          const bridge = new TransformStream({
+            flush() {
+              messageCallbacks.delete(id);
+            },
+          });
+          mesg.stream.pipeThrough(bridge);
+          resolve([bridge.readable, mesg.context as Context]);
```

Comment:
I didn't know this `.readable` usage.
I would have done it like this, but either should be fine:
```suggestion
          resolve([mesg.stream.pipeThrough(bridge), mesg.context as Context]);
```